lib/hashid/rails.rb

```diff
@@ -4,22 +4,34 @@
 
 module Hashid
   module Rails
-    def self.included(base)
-      base.extend ClassMethods
-    end
+    def hashid(options = {})
+      extend ClassMethods
+      include InstanceMethods
 
-    def encoded_id
-      self.class.encode_id(id)
+      cattr_accessor :hashid_options
+      self.hashid_options = ({length: 6}.merge(options))
     end
 
-    def to_param
-      encoded_id
+    module InstanceMethods
+      def encoded_id
+        self.class.encode_id(id)
+      end
+
+      def to_param
+        encoded_id
+      end
+      alias_method :hashid, :to_param
     end
-    alias_method :hashid, :to_param
 
     module ClassMethods
       def hashids
-        Hashids.new(table_name, 6)
+        salt = if self.hashid_options[:secure]
+                 "#{table_name}:#{::Rails.application.secrets.secret_key_base}"
```

Comment:
I'm considering using `ENV['HASHIDS_SALT']` - the reason being, if I ever need to reset my secret_key_base, I will not invalidate all my HashIDs. I also feel slightly uncomfortable using my secret key in this way, because I know HashIDs aren't designed for security, and - although I doubt it's feasible - I can imagine someone reverse engineering my `secret_key_base` if they had a set of my HashIDs from which to brute force it.